HW2 - Async development

wenditer opened the pull request:
- monitor 'data' folder for changes and import csv files to json

wenditer committed:
HW2 - Async development

- monitor 'data' folder for changes and import csv files to json

abruzgin commented on src/app.js:
```diff
@@ -1,8 +1,24 @@
+
+console.log("---> HW2 - Async development");
+console.log("Watching for changes in 'data' directory and import CSV files as json:");
+import DirWatcher from "./csv/dirwatcher"
```
Basically in JS development, I guess the same like in Java or any other language, modules declarations should go first. So please combine them in one place

wenditer replied:
Usually in Java I created a single responsible modules/classes and not a bunch of homeworks =) Here I divided different homeworks to a kind of groups, every group starts with console.log("---> HWX - description"); and contains all the stuff needed for single homework in one place. May be it would be better to not mix HWs, but extract them into separate modules? e.g. create hw1.js, hw2.js or some kind of this.

abruzgin commented on src/csv/dirwatcher.js:
```diff
@@ -0,0 +1,48 @@
+import EventEmitter from 'events'
+import Set from 'collections/set'
```
Actually there's a native implementation of Set
https://developer.mozilla.org/ru/docs/Web/JavaScript/Reference/Global_Objects/Set

wenditer replied:
This implementation was more convenient, it has methods difference, filter, map etc. I'd prefer to keep it as is if there is no simple analogues of these methods in native Set.

abruzgin commented on src/csv/dirwatcher.js:
```diff
@@ -0,0 +1,48 @@
+    this.watchInterval = setInterval(() => {
+      readdir(path)
+        .then((filesFromFS) => {
+          const newFiles = new Set(filesFromFS).difference(this.processedFiles);
```
For managing Sets and other data structures you can use `lodash` library

wenditer replied:
Thanks. Will try it in next HWs. For this HW Set methods were extremely easy and convenient to use, so I didn't try to find any alternatives.

abruzgin commented on src/csv/dirwatcher.js:
```diff
@@ -0,0 +1,48 @@
+        })
+        .catch((error) => {
+          console.log(error);
+          throw error;
```
For error handling in JS devs use to use `throw new Error()`

abruzgin commented on src/csv/importer.js:
```diff
@@ -0,0 +1,32 @@
+    return csv().fromFile(path);
+  }
+
+  async importPseudoSync(path) {
```
What does it stand for?

wenditer committed:
HW2 - Async development

- fix code review comments
  - divide app.js into single responsible files and do imports at the very top of file
  - fix throw error

wenditer committed:
HW2 - Async development

- fix code review comments - get rid of collections, add lodash
